### PR TITLE
Number buttons running with MVP arc.

### PR DESCRIPTION
**Pull request by Matias-Villalobo**
Now the number buttons are running, following the MVP flow. Note: The application nowadays does not do any calculations, it just only saves and displays the numbers that the user click. In a future Pr i will add the sum operation.

WhatsApp Video 2020-12-17 at 14 30 33

### CODE REVIEW

**Comment by LuchoElissondoGlobant** [comment]
for next pr you can do this
https://developer.android.com/studio/debug/am-video

File: app/src/main/java/com/example/mycalculator/MainActivity.java [hunk 1]
```diff
@@ -2,13 +2,99 @@
...
-        setContentView(R.layout.activity_main);
+        binding = ActivityMainBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        presenter = new CalculatorPresenterImpl(new CalculatorModelImpl(), new CalculatorViewImpl(this, binding));
+
+        binding.buttonNumberZero.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.numberPressed("0");
```
Comment (BQuinteros) [inline app/src/main/java/com/example/mycalculator/MainActivity.java:36]: pls remove the hardcode, you can move this to a Util class, create a class with all constant.

File: app/src/main/java/com/example/mycalculator/MainActivity.java [hunk 2]
```diff
@@ -2,13 +2,99 @@
...
+import com.example.mycalculator.mvp.CalculatorPresenterImpl;
+import com.example.mycalculator.mvp.CalculatorViewImpl;
 
 public class MainActivity extends AppCompatActivity {
 
+
+    private ActivityMainBinding binding;
+    private CalculatorPresenterImpl presenter;
+
+
```
Comment (LuchoElissondoGlobant) [inline app/src/main/java/com/example/mycalculator/MainActivity.java:20]: remove extra liness please

File: app/src/main/java/com/example/mycalculator/MainActivity.java [hunk 3]
```diff
@@ -2,13 +2,99 @@
...
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_main);
+        binding = ActivityMainBinding.inflate(getLayoutInflater());
+        setContentView(binding.getRoot());
+
+        presenter = new CalculatorPresenterImpl(new CalculatorModelImpl(), new CalculatorViewImpl(this, binding));
+
+        binding.buttonNumberZero.setOnClickListener(new View.OnClickListener() {
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/MainActivity.java:33]: put all the **setOnClickListener** in a separated method to grant more legibility. Maybe a method called setListeners or something like that

File: app/src/main/java/com/example/mycalculator/MainActivity.java (hunk 1 above)
Comment (LuchoElissondoGlobant) [inline app/src/main/java/com/example/mycalculator/MainActivity.java:36]: dont use harcodes please. you shoud use string resource

Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/MainActivity.java]: here you are in the activity so you can use the string resources instead of hardcode. Please use them

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorContract.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package com.example.mycalculator.mvp;
+
+public interface CalculatorContract {
```
Comment (BQuinteros) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorContract.java]: Nicee!!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+    @Override
+    public void saveNumber(String number) {
+        operatorOne = number;
+    }
+
+    public String getResult() {
+        return operatorOne;
+    }
+
```
Comment (BQuinteros) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java:18]: Remove extra empty lines

File: app/src/main/java/com/example/mycalculator/MainActivity.java [hunk 6]
```diff
@@ -2,13 +2,99 @@
...
+import com.example.mycalculator.databinding.ActivityMainBinding;
+import com.example.mycalculator.mvp.CalculatorModelImpl;
+import com.example.mycalculator.mvp.CalculatorPresenterImpl;
+import com.example.mycalculator.mvp.CalculatorViewImpl;
 
 public class MainActivity extends AppCompatActivity {
 
+
+    private ActivityMainBinding binding;
+    private CalculatorPresenterImpl presenter;
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/MainActivity.java]: the presenter type must be the contract

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
```
Comment (LuchoElissondoGlobant) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java]: remove extra lines

File: app/src/main/java/com/example/mycalculator/mvp/ActivityView.java [hunk 8]
```diff
@@ -0,0 +1,4 @@
+package com.example.mycalculator.mvp;
+
+public class ActivityView {
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/ActivityView.java]: why this class is empty?

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+    }
+
+    @Override
+    public void numberPressed(String number) {
+
+        model.saveNumber(number);
+        view.drawNumber(model.getResult());
+    }
+
+
```
Comment (BQuinteros) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java:21]: Remove empty lines pls

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorContract.java [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+    public interface calculatorPresenter {
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorContract.java]: maybe you can change the name so the contract will be called **CalculatorPresenterContract** and the implementation **CalculatorPresenter**.

Comment (SartiMau) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorContract.java]: All the class names should start with upper case

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java (hunk 5 above)
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java]: remove this extra space pls

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java [hunk 11]
```diff
@@ -0,0 +1,22 @@
+package com.example.mycalculator.mvp;
+
+public class CalculatorPresenterImpl implements CalculatorContract.calculatorPresenter {
+
+    private CalculatorModelImpl model;
+    private CalculatorViewImpl view;
+
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java]: remove this extra space pls

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+        this.view = view;
+    }
+
+    @Override
+    public void numberPressed(String number) {
+
+        model.saveNumber(number);
+        view.drawNumber(model.getResult());
+    }
+
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java]: remove this extra space pls

File: app/src/main/java/com/example/mycalculator/mvp/ActivityView.java [hunk 13]
```diff
@@ -0,0 +1,4 @@
...
+}
```
Comment (LuchoElissondoGlobant) [inline app/src/main/java/com/example/mycalculator/mvp/ActivityView.java]: you should delete this empty class

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java [hunk 14]
```diff
@@ -0,0 +1,22 @@
+package com.example.mycalculator.mvp;
+
+public class CalculatorPresenterImpl implements CalculatorContract.calculatorPresenter {
+
+    private CalculatorModelImpl model;
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java]: the type should be the interface

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+    private CalculatorViewImpl view;
+
+
+    public CalculatorPresenterImpl(CalculatorModelImpl model, CalculatorViewImpl view) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java]: same here, the type should be the interface

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorViewImpl.java [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+import android.widget.TextView;
+
+import com.example.mycalculator.MainActivity;
+import com.example.mycalculator.databinding.ActivityMainBinding;
+
+public class CalculatorViewImpl implements CalculatorContract.calculatorView {
+
+    private ActivityMainBinding binding;
+
+    public CalculatorViewImpl(MainActivity mainActivity, ActivityMainBinding binding) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorViewImpl.java]: you are not using the **mainActivity** param, you can delete it

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/mycalculator/MainActivity.java (hunk 1 above)
Comment (SartiMau) [reply app/src/main/java/com/example/mycalculator/MainActivity.java]: to be more clear, you can use binding.buttonNumberZero.text

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java [hunk 17]
```diff
@@ -0,0 +1,20 @@
+package com.example.mycalculator.mvp;
+
+public class CalculatorModelImpl implements CalculatorContract.calculatorModel {
+
+    public String getResult;
+    private String operatorOne;
```
Comment (mauroLonghi) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java]: no quedaria mejor ponerle firstOperator & secondOperator

File: app/src/main/java/com/example/mycalculator/MainActivity.java (hunk 2 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/MainActivity.java:20]: Lines removed. Thx

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java (hunk 7 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java]: Lines removed. Thx

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorContract.java (hunk 4 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorContract.java]: Thx!

File: app/src/main/java/com/example/mycalculator/mvp/ActivityView.java (hunk 13 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/ActivityView.java]: After the meeting i had with Lucho and Mau, they told me about the weak references, and how to use the ActivityView, more efficiently. Thx!!

File: app/src/main/java/com/example/mycalculator/MainActivity.java (hunk 3 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/MainActivity.java:33]: Done, now they are all inside a method call setListeners(). Thx!!

File: app/src/main/java/com/example/mycalculator/MainActivity.java (hunk 1 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/MainActivity.java]: Done. They are not hardcoded anymore., now the code is (exple):  "presenter.numberPressed((String) binding.buttonNumberZero.getText());"

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java (hunk 17 above)
Comment (SartiMau) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java]: sure! firstOperand, secondOperand and operator

File: app/src/main/java/com/example/mycalculator/mvp/ActivityView.java (hunk 8 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/ActivityView.java]: After the meeting i had with Lucho and Mau, they told me about the weak references, and how to use the ActivityView, more efficiently. Thx!!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorContract.java (hunk 10 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorContract.java]: Done. Thx!!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java (hunk 11 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java]: Can you specify which space?

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java (hunk 12 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java]: Done. Thx!!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorViewImpl.java (hunk 16 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorViewImpl.java]: Done. Thx!!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java (hunk 17 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java]: Done. You were right thx!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java [hunk 18]
```diff
@@ -0,0 +1,20 @@
+package com.example.mycalculator.mvp;
+
+public class CalculatorModelImpl implements CalculatorContract.calculatorModel {
+
+    public String getResult;
```
Comment (LuchoElissondoGlobant) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java]: unused property here. please delete

Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java]: Done. Thx!!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java (hunk 5 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java]: Done. Thx!!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java (hunk 11 above)
Comment (LuchoElissondoGlobant) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java]: he meant extra line. 7 or 8

Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java]: Thx for the tip!!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java (hunk 14 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java]: Done. Thx!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java (hunk 15 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java]: Done. Thx!

File: app/src/main/java/com/example/mycalculator/MainActivity.java (hunk 6 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/MainActivity.java]: Done. Thx!

File: app/src/main/java/com/example/mycalculator/MainActivity.java [hunk 19]
```diff
@@ -2,13 +2,99 @@
...
+        presenter = new CalculatorPresenterImpl(new CalculatorModelImpl(), new CalculatorViewImpl(this, binding));
+
+        setListeners();
+    }
+
+    private void setListeners() {
+        binding.buttonNumberZero.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.numberPressed((String) binding.buttonNumberZero.getText().toString());
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/MainActivity.java]: as you call the **.toString()** the String cast is not necessary anymore. Please remove it

File: app/src/main/java/com/example/mycalculator/MainActivity.java [hunk 20]
```diff
@@ -2,13 +2,99 @@
...
+        binding.buttonNumberNine.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.numberPressed((String) binding.buttonNumberNine.getText().toString());
+            }
+        });
+        binding.buttonFunctionErase.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                presenter.numberPressed("");
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/MainActivity.java]: no hardcode pls, create a StringUtils class and put there a EMPTY_STRING constant

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java [hunk 21]
```diff
@@ -0,0 +1,17 @@
+package com.example.mycalculator.mvp;
+
+public class CalculatorModelImpl implements CalculatorContract.CalculatorModelContract {
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java]: you can change the name of this class and call it **CalculatorModel**

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java [hunk 22]
```diff
@@ -0,0 +1,17 @@
...
+
+    private String firstOperator;
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java]: this should call firstOperand as will be the first number not the operator

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java [hunk 23]
```diff
@@ -0,0 +1,17 @@
...
+    private String secondOperator;
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java]: you are not using this variable, pls remove it

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java [hunk 24]
```diff
@@ -0,0 +1,17 @@
...
+    private String operation;
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java]: same

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java [hunk 25]
```diff
@@ -0,0 +1,17 @@
+package com.example.mycalculator.mvp;
+
+public class CalculatorPresenterImpl implements CalculatorContract.CalculatorPresenterContract {
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java]: you can change the name of this class and call it CalculatorPresenter

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java [hunk 26]
```diff
@@ -0,0 +1,17 @@
...
+
+    private CalculatorContract.CalculatorModelContract model;
+    private CalculatorContract.CalculatorViewContract view;
+
+    public CalculatorPresenterImpl(CalculatorModelImpl model, CalculatorViewImpl view) {
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java]: here maybe you can specify the contract type in the paremeters

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorViewImpl.java [hunk 27]
```diff
@@ -0,0 +1,19 @@
+package com.example.mycalculator.mvp;
+
+import com.example.mycalculator.MainActivity;
+import com.example.mycalculator.databinding.ActivityMainBinding;
+
+public class CalculatorViewImpl extends ActivityView implements CalculatorContract.CalculatorViewContract {
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorViewImpl.java]: you can change the name of this class and call it CalculatorView

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorViewImpl.java (hunk 27 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorViewImpl.java]: Done. Thx!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java (hunk 25 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java]: Done. Thx!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java (hunk 24 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java]: Done. Thx!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java (hunk 23 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java]: Done. Thx!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java (hunk 22 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java]: Done. Thx!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java (hunk 21 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorModelImpl.java]: Done. Thx!

File: app/src/main/java/com/example/mycalculator/MainActivity.java (hunk 19 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/MainActivity.java]: Done. Thx!

File: app/src/main/java/com/example/mycalculator/MainActivity.java (hunk 20 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/MainActivity.java]: Done. Thx!!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java (hunk 26 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenterImpl.java]: Done. Thx!

File: app/src/main/java/com/example/mycalculator/StringUtils.java [hunk 28]
```diff
@@ -0,0 +1,6 @@
+package com.example.mycalculator;
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/StringUtils.java]: yo can create a package called utils, and put this class inside

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorContract.java [hunk 29]
```diff
@@ -0,0 +1,18 @@
...
+public interface CalculatorContract {
+
+    public interface CalculatorPresenterContract {
+        public void numberPressed(String number);
+    }
+
+    public interface CalculatorModelContract {
+        public void saveNumber(String number);
+
+        String getResult();
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorContract.java:12]: do not put variables in the contracts

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorContract.java [hunk 30]
```diff
@@ -0,0 +1,18 @@
+package com.example.mycalculator.mvp;
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorContract.java]: you can create a package called contract inside mvp, and put this class inside

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModel.java [hunk 31]
```diff
@@ -0,0 +1,15 @@
+package com.example.mycalculator.mvp;
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorModel.java]: yo can create a package called model, and put this class inside

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenter.java [hunk 32]
```diff
@@ -0,0 +1,18 @@
+package com.example.mycalculator.mvp;
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenter.java]: yo can create a package called presenter, and put this class inside

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorView.java [hunk 33]
```diff
@@ -0,0 +1,19 @@
+package com.example.mycalculator.mvp;
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/mvp/CalculatorView.java]: yo can create a package called view, and put this class inside

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorView.java (hunk 33 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorView.java]: Done. Thx!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenter.java (hunk 32 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorPresenter.java]: Done. Thx!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorModel.java (hunk 31 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorModel.java]: Done. Thx!

File: app/src/main/java/com/example/mycalculator/mvp/CalculatorContract.java (hunk 30 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/mvp/CalculatorContract.java]: Done. Thx!

File: app/src/main/java/com/example/mycalculator/StringUtils.java (hunk 28 above)
Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/StringUtils.java]: Done. Thx!

File: app/src/main/java/com/example/mycalculator/MainActivity.java [hunk 34]
```diff
@@ -2,13 +2,100 @@
...
+import com.example.mycalculator.mvp.presenter.CalculatorPresenter;
+import com.example.mycalculator.mvp.view.CalculatorView;
+import com.example.mycalculator.utils.StringUtils;
 
 public class MainActivity extends AppCompatActivity {
 
+    private ActivityMainBinding binding;
+    private CalculatorContract.CalculatorPresenterContract presenter;
+
+    @SuppressLint("WrongViewCast")
```
Comment (SartiMau) [inline app/src/main/java/com/example/mycalculator/MainActivity.java]: why did you add this?

Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/MainActivity.java]: @SuppressLint("WrongViewCast") this?

Comment (SartiMau) [reply app/src/main/java/com/example/mycalculator/MainActivity.java]: yes

Comment (Matias-Villalobo) [reply app/src/main/java/com/example/mycalculator/MainActivity.java]: If i remeber correctly, when i was doing the calculator layout, i think i got an error of the lint tool, and to solve that problem i used a quick fix with alt*enter. Regardless of that, following your comment, i erased it and now it´s no more, and the app still works.

Comment (SartiMau) [reply app/src/main/java/com/example/mycalculator/MainActivity.java:21]: okay, try to remove it ;)

**Review by LuchoElissondoGlobant — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Review by BQuinteros — APPROVED** [review]
